subsys/portability/cmsis_rtos_v1/cmsis_thread.c

```diff
@@ -29,6 +34,8 @@ static inline uint32_t cmsis_to_zephyr_priority(int32_t c_prio)
 
 static void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
 {
+	k_thread_custom_data_set(arg2);
```

Comment:
The application can set custom data and overrides this. So this cannot be used to store instance number.